feat: update Kotlin 2.0.20-RC1 release details (#4344)
JetBrains/kotlin-web-site: docs/topics/whatsnew-eap.md

If you still use earlier versions of Xcode but want to upgrade to %kotlinEapVersion%, 
disable bitcode embedding in your Xcode projects.

### Changes to monitoring GC performance with signposts

> Kotlin %kotlinEapVersion% made it possible to monitor the performance of Kotlin/Native garbage collector 

Reviewer: ```suggestion
Kotlin %kotlinEapVersion% makes it possible to monitor the performance of the Kotlin/Native garbage collector 
```

For more information, see the issue in [YouTrack](https://youtrack.jetbrains.com/issue/KT-69330).

### Option to share JVM artifacts between projects as class files

> > This feature is Experimental. It may be dropped or changed at any time. Use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack]()https://youtrack.jetbrains.com/issue/KT-61861/Gradle-Kotlin-compilations-depend-on-packed-artifacts. Opt-in is required (see details below).

Reviewer: ```suggestion
> This feature is [Experimental](components-stability.md#stability-levels-explained). It may be dropped or changed at any time. Use it only for evaluation purposes. We would appreciate your feedback on it in [YouTrack](https://youtrack.jetbrains.com/issue/KT-61861/Gradle-Kotlin-compilations-depend-on-packed-artifacts). Opt-in is required (see details below).
```

>
{type="warning"}

> In Kotlin %kotlinEapVersion%, we introduce a new approach that changes the way the outputs of Kotlin/JVM compilations, such as JAR files, are shared between projects. With this approach, Gradle’s `apiElements` configuration now has a secondary variant that provides access to the directory containing compiled `.class` files. When configured, your project uses this directory instead of requesting the compressed JAR artifact during compilation. This reduces the number of times JAR files are compressed and decompressed, especially for incremental builds.

Reviewer: ```suggestion
In Kotlin %kotlinEapVersion%, we introduce a new approach that changes the way the outputs of Kotlin/JVM compilations, such as JAR files, are shared between projects. With this approach, Gradle's `apiElements` configuration now has a secondary variant that provides access to the directory containing compiled `.class` files. When configured, your project uses this directory instead of requesting the compressed JAR artifact during compilation. This reduces the number of times JAR files are compressed and decompressed, especially for incremental builds.
```

Our testing shows that this new approach can provide build performance improvements for Linux and macOS hosts. However, on Windows hosts, we have seen a degradation in performance due to how Windows handles I/O operations when working with files.

To try this new approach, add the following Gradle property to your `gradle.properties` file:

```none
kotlin.jvm.addClassesVariant=true
```

By default, this property is set to false and the `apiElements` variant in Gradle requests the compressed JAR artifact.

> Gradle has a related property that you can use in your Java-only projects to only expose the  
> compressed JAR artifact during compilation **instead** of the directories containing compiled
> `.class` files:
>
> ```none
> org.gradle.java.compile-classpath-packaging=true
> ```
>

> > For more information on this property and its purpose, see Gradle’s documentation on [Significant build performance drop on Windows for huge multi-projects](https://docs.gradle.org/current/userguide/java_library_plugin.html#sub:java_library_known_issues_windows_performance).

Reviewer: ```suggestion
> For more information on this property and its purpose, see Gradle's documentation on [Significant build performance drop on Windows for huge multi-projects](https://docs.gradle.org/current/userguide/java_library_plugin.html#sub:java_library_known_issues_windows_performance).
```

>
{type="note"}

We would appreciate your feedback on this new approach. Do you also see performance improvements? Let us know by adding a comment in [YouTrack](https://youtrack.jetbrains.com/issue/KT-61861/Gradle-Kotlin-compilations-depend-on-packed-artifacts).

> ### Compose compiler plugin updates ✅

Reviewer: ```suggestion
### Compose compiler plugin updates
```

In %kotlinEapVersion%, the Compose compiler Gradle plugin gets a few improvements.

#### New way to configure compiler options

We have introduced a new option configuration mechanism to avoid churn of top-level parameters: 

> it’s harder for the Compose compiler team to test things out 

Reviewer: ```suggestion
it's harder for the Compose compiler team to test things out 
```
